service/src/main/java/bio/terra/workspace/service/resource/controlled/cloud/aws/storageFolder/DeleteAwsStorageFolderStep.java

```diff
@@ -0,0 +1,53 @@
+package bio.terra.workspace.service.resource.controlled.cloud.aws.storageFolder;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.common.utils.AwsUtils;
+import bio.terra.workspace.service.workspace.AwsCloudContextService;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+
+public class DeleteAwsStorageFolderStep implements Step {
+  private static final Logger logger = LoggerFactory.getLogger(DeleteAwsStorageFolderStep.class);
+  private final ControlledAwsStorageFolderResource resource;
+  private final AwsCloudContextService awsCloudContextService;
+
+  public DeleteAwsStorageFolderStep(
+      ControlledAwsStorageFolderResource resource, AwsCloudContextService awsCloudContextService) {
+    this.resource = resource;
+    this.awsCloudContextService = awsCloudContextService;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    AwsCredentialsProvider credentialsProvider =
+        AwsUtils.createWsmCredentialProvider(
+            awsCloudContextService.getRequiredAuthentication(),
+            awsCloudContextService.discoverEnvironment());
+
+    // TODO(TERRA-279): check permissions to delete
```

Comment:
We already checked this permission in the controller

Reply:
Retaining this since AWS account / landing zone handling 'may' be different